File: app/src/main/java/com/example/shibegallery/viewmodel/MainViewModel.java
package com.example.shibegallery.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.shibegallery.repository.ShibeRepo;

import org.jetbrains.annotations.NotNull;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainViewModel extends ViewModel {

    private MutableLiveData<List<String>> _imageUrls = new MutableLiveData<>();

    public LiveData<List<String>> getImageUrls() {
        return _imageUrls;
    }

    public void fetchShibeImageUrls(String path, int count) {
        new ShibeRepo().getImageUrls(path, count).enqueue(new Callback<List<String>>() {
            @Override
            public void onResponse(@NotNull Call<List<String>> call, @NotNull Response<List<String>> response) {
                List<String> urls = response.body();
                _imageUrls.setValue(urls);
            }

            @Override
            public void onFailure(@NotNull Call<List<String>> call, @NotNull Throwable t) {
                // TODO: 7/31/2020 Handle Error
            }
        });
    }
}

File: appkotlin/src/main/java/com/example/appkotlin/viewmodel/MainViewModel.kt
package com.example.appkotlin.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.example.appkotlin.repository.Repo
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainViewModel : ViewModel() {

    private val _urls = MutableLiveData<List<String>>()
    val urlObservable: LiveData<List<String>>
        get() = _urls

    fun fetchImages(animal: String, count: Int) {
        Repo().callListOfString(animal, count).enqueue(object : Callback<List<String>> {
            override fun onFailure(call: Call<List<String>>, t: Throwable) {
                TODO("Not yet implemented")
            }

            override fun onResponse(call: Call<List<String>>, response: Response<List<String>>) {
                _urls.value = response.body()
            }

        })
    }
}